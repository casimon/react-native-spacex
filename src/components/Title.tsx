import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';

const styles = StyleSheet.create({
  title: {
    flex: 1,
  },
  titleText: {
    fontSize: 16,
    color: '#000000',
  },
});

interface Props {
  text: string;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
}

class Title extends React.Component<Props> {
  render() {
    return (
      <View style={[styles.title, this.props.style]}>
        <Text style={[styles.titleText, this.props.styleText]}>
          {this.props.text}
        </Text>
      </View>
    );
  }
}

export default Title;
