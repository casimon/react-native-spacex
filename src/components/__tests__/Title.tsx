/**
 * @format
 */

import 'react-native';
import React from 'react';
import Title from '../Title';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders title with text', () => {
  const tree = renderer.create(<Title text="Test text prop" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders title with text and optional style prop', () => {
  const tree = renderer
    .create(<Title text="Test text" style={{margin: 2}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders title with text and optional styleText prop', () => {
  const tree = renderer
    .create(<Title text="Test text" styleText={{fontSize: 12}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
