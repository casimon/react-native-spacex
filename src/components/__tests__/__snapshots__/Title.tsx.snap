// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title with text 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Test text prop
  </Text>
</View>
`;

exports[`renders title with text and optional style prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      Object {
        "margin": 2,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Test text
  </Text>
</View>
`;

exports[`renders title with text and optional styleText prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 3,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
        },
        Object {
          "fontSize": 12,
        },
      ]
    }
  >
    Test text
  </Text>
</View>
`;
